src/mongocxx/options/count.cpp

```diff
@@ -33,6 +36,8 @@ count& count::limit(std::int64_t limit) {
 }
 
 count& count::max_time(std::chrono::milliseconds max_time) {
+    if (!is_int32_duration<std::chrono::milliseconds>(max_time))
```

Comment:
I agree that the problem is when they are applied over *there*, but the point at which the user can do anything about it is when the options are set.  I think it's more user friendly to raise the exception early rather than when the option gets used, which would require the user to backtrack to find where the option was set anyway.